src/test/java/it/gov/pagopa/payhub/auth/controller/AuthzControllerNoOrganizzationAccessModeTest.java

```diff
@@ -211,4 +214,81 @@ void givenAuthorizedUserWhenRegisterClientThenOk() throws Exception {
         assertEquals(uuidRandomForSecret, clientDTO.getClientSecret());
     }
 
+    @Test
+    void givenAuthorizedUserWhenGetClientSecretThenOk() throws Exception {
+        String uuidRandomForClientSecret = UUID.randomUUID().toString();
+        String organizationIpaCode = "IPA_TEST_2";
+        String clientId = "CLIENTID";
+
+        UserInfo expectedUser = UserInfo.builder()
+          .userId("USERID")
+          .organizationAccess(organizationIpaCode)
+          .organizations(List.of(UserOrganizationRoles.builder()
+            .organizationIpaCode(organizationIpaCode)
+            .roles(List.of(Constants.ROLE_ADMIN))
+            .build()))
+          .build();
+
+        Mockito.when(authnServiceMock.getUserInfo("accessToken"))
+          .thenReturn(expectedUser);
+
+        doReturn(uuidRandomForClientSecret)
+          .when(authzServiceMock).getClientSecret(organizationIpaCode, clientId);
+
+        MvcResult result = mockMvc.perform(
+            get("/payhub/auth/clients/{organizationIpaCode}/{clientId}", organizationIpaCode, clientId)
+              .header(HttpHeaders.AUTHORIZATION, "Bearer accessToken")
+          ).andExpect(status().isOk())
+          .andReturn();
+
+        assertEquals(uuidRandomForClientSecret, result.getResponse().getContentAsString());
+    }
+
+    @Test
+    void givenAuthorizedUserWhenGetClientsThenOk() throws Exception {
```

Comment:
let's test for both the API the not authorized outcome

Reply:
you are introducing the ClientNotFoundException, what happen if the service throws it?